source/js/buyers-guide/search.js

```diff
@@ -15,10 +15,63 @@ const categoryTitle = document.querySelector(`.category-title`);
 const parentTitle = document.querySelector(`.parent-title`);
 const toggle = document.querySelector(`#product-filter-pni-toggle`);
 const subcategories = document.querySelectorAll(`.subcategories`);
+const subContainer = document.querySelector(`.subcategory-header`);
 
 // TODO: turn this into a static class rather than plain JS object.
 const SearchFilter = {
   init: () => {
+    let pos = { left: 0, x: 0 };
+
+    const markScrollStart = (event) => {
+      event.preventDefault();
+      event.stopImmediatePropagation();
+      subContainer.classList.add("cursor-grabbing");
+      subContainer.classList.add("select-none");
+
+      pos = {
+        left: subContainer.scrollLeft,
+        x: event.clientX,
+      };
+
+      document.addEventListener("mousemove", markScrollMove);
+      document.addEventListener("mouseup", markScrollEnd);
+      document.addEventListener("touchmove", markScrollMove);
+      document.addEventListener("touchend", markScrollEnd);
+      document.addEventListener("touchcancel", markScrollEnd);
```

Comment:
let's compact this into only two `addListener` calls, but for lists of event names. That way, there's fewer places the event binding can go wrong, and updating the events is just a matter of adding/removing event names:

```suggestion
      [`mousemove`, `touchmove`].forEach(type =>
        document.addEventListener(type, markScrollMove)
      );
            
      [`mouseup`, `touchend`, `touchcancel`].forEach(type =>
        document.addEventListener(type, markScrollEnd)
      );
```